doc/porting_guide.md

```diff
@@ -64,24 +64,30 @@ BSP Functions are broken down by logical subsystems as follows:
 
 ### BSP Modules
 
+#### Optional
+- BSP FWU: Firmware Update Inegration for Secure File Transfer (SFT) (`include/bsp/xi_bsp_fwu.h`)
+- BSP IO FS: File System Integration for SFT (`include/bsp/xi_bsp_io_fs.h`)
+
+#### Required
 - BSP IO NET: Networking Stack Integration (`include/bsp/xi_bsp_io_net.h`)
 - BSP TLS: Transport Layer Security Integration (`include/bsp/xi_bsp_tls.h`)
 - BSP MEM: Heap Memory Management (`include/bsp/xi_bsp_mem.h`)
 - BSP RNG: Random Number Generator (`include/bsp/xi_bsp_rng.h`)
 - BSP TIME: Time Function (`include/bsp/xi_bsp_time.h`)
 
+
 ### Reference Implementations
 
 Reference **function implementations** are separated into two directories: `src/bsp/platform` and `src/bsp/tls`.
 
-The *platform* directory contains networking, memory, random number generator and time implementations.  The *tls* directory contains reference implementations for *wolfSSL* and *mbedTLS* libraries which supply secure TLS v1.2 connections over TCP/IP.
+The *platform* directory contains SFT, networking, memory, random number generator and time implementations.  The *tls* directory contains reference implementations for *wolfSSL* and *mbedTLS* libraries which supply secure TLS v1.2 connections over TCP/IP.
```

Comment:
again, order, separation and phrasing: networking, file io, memory, ... , firmware management.